src/plugins/prefs/README.md

```diff
@@ -0,0 +1,107 @@
+- infos = Information about the prefs plugin is in keys below
+- infos/author = Name <name@libelektra.org>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides = storage
+- infos/recommends =
+- infos/placements = getstorage setstorage
+- infos/status = maintained reviewed conformant compatible coverage specific tested nodep libc preview experimental difficult unfinished nodoc concept 
+- infos/metadata =
+- infos/description =
+
+## Basics ##
+
+### Preference Types ###
+
+- Default preferences: below `mountpoint/preferences/pref/`.
+- User preferences: below `mountpoint/preferences/user/`.
+- Lock preferences: below `mountpoint/preferences/lock/`.
+
+User preferences will override default preferences. Lock preferences can't be overwritten by user or default preferences. 
+Only user preferences will be saved to `prefs.js`.
```

Comment:
that part is a little bit hard to explain, there are 4 different kind of preferences. 
user preferences are usually set (manually) by the user via about:config and will be saved to/read from the users firefox config file `~/.mozilla/firefox/<profile_dir>/prefs.js`. it's not the storage plugin, but the autoconfig script that automatically saves the user preferences to `prefs.js` because otherwise preferences set by the user manually while we intercept would be lost when shutting down firefox. 
a side effect is that automatically generated *user* preferences will be saved too, but that shouldn't be an issue.

default and lock preferences are "read-only" (usually located in `/usr/lib/firefox`, `/etc/firefox`, `/usr/share/firefox`, ...) and they won't be saved to any preference file.